src/app.js

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2015 Colin Clark
+
+Licensed under the 3-Clause "New" BSD license.
+You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the 3-Clause "New" BSD License at
+https://github.com/colinbdclark/infusion-electron/raw/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    electron = fluid.registerNamespace("electron");
+
+// Monkey-patch fluid.log for Electron's broken console.log.
+fluid.doLog = function (args) {
+    if (typeof (console) !== "undefined") {
+        if (console.debug) {
+            console.debug(args.join(" "));
+        } else if (typeof (console.log) === "function") {
+            console.log(args.join(" "));
+        }
+    }
+};
+
+
+fluid.defaults("electron.app", {
+    gradeNames: "fluid.modelComponent",
+
+    members: {
+        app: require("app")
+    },
+
+    commandLineSwitches: {},
+
+    windowListeners: {
+        "onClose": "electron.app.quitOnAllClosed()"
+    },
+
+    events: {
+        onReady: null,
+        onAllWindowsClosed: null
+    },
+
+    listeners: {
+        onCreate: [
+            {
+                funcName: "electron.app.setCommandLineSwitches",
+                args: ["{that}.app", "{that}.options.commandLineSwitches"]
+            },
+            {
+                "this": "{that}.app",
+                method: "on",
+                args: ["window-all-closed", "{that}.events.onAllWindowsClosed.fire"]
+            },
+            {
+                "this": "{that}.app",
+                method: "on",
+                args: ["ready", "{that}.events.onReady.fire"]
+            }
+        ],
+
+        onAllWindowsClosed: [
```

Comment:
I can't find anything in the documentation that suggests we need to clean any resources up.

https://github.com/atom/electron/blob/master/docs/api/app.md

It might make sense to establish a contract where <code>quit()</code> is called if the app component is destroyed, but I'm not yet familiar enough with Electron to know how this should work.